controllers/challenge.go

```diff
@@ -16,6 +17,13 @@ func GetChallengeByID(ctx context.Context, w http.ResponseWriter, r *http.Reques
 		return
 	}
 
+	p, err := passenger.FromRequest(ctx, r)
```

Comment:
At this point you should call `passenger.FromContext` to avoid querying Datastore again.